From 6570337948b5dd25e20cb7e6db45a3abb5b62d39 Mon Sep 17 00:00:00 2001
From: qneill <qneill@138bc75d-0d04-0410-961f-82ee72b054a4>
Date: Tue, 31 May 2011 14:32:38 +0000
Subject: [PATCH] Fix fma4_fmsubadd and fma4_fmaddsub mode suffixes.

2011-05-31  Quentin Neill  <quentin.neill@amd.com>

	* config/i386/sse.md (fma4_fmsubadd): Use <ssemodesuffix>.
	(fma4_fmaddsub): Likewise

git-svn-id: svn+ssh://gcc.gnu.org/svn/gcc/branches/gcc-4_6-branch@174485 138bc75d-0d04-0410-961f-82ee72b054a4

index 7e18678..0e03dd4 100644
--- a/gcc/config/i386/sse.md
+++ b/gcc/config/i386/sse.md
@@ -2068,7 +2068,7 @@
 	   (match_operand:AVXMODEF2P 3 "nonimmediate_operand" "xm,x")]
 	  UNSPEC_FMADDSUB))]
   "TARGET_FMA4"
-  "vfmaddsubps\t{%3, %2, %1, %0|%0, %1, %2, %3}"
+  "vfmaddsub<ssemodesuffix>\t{%3, %2, %1, %0|%0, %1, %2, %3}"
   [(set_attr "type" "ssemuladd")
    (set_attr "mode" "<MODE>")])
 
@@ -2081,7 +2081,7 @@
 	     (match_operand:AVXMODEF2P 3 "nonimmediate_operand" "xm,x"))]
 	  UNSPEC_FMADDSUB))]
   "TARGET_FMA4"
-  "vfmsubaddps\t{%3, %2, %1, %0|%0, %1, %2, %3}"
+  "vfmsubadd<ssemodesuffix>\t{%3, %2, %1, %0|%0, %1, %2, %3}"
   [(set_attr "type" "ssemuladd")
    (set_attr "mode" "<MODE>")])
 
-- 
1.7.0.4

